app/controllers/rewards_controller.rb

```diff
@@ -1,9 +1,23 @@
 class RewardsController < EmployeeBaseController
   def index
-    @rewards = Reward.page(params[:page])
+    @rewards = if reward_params.key?('category')
+                 Reward.includes(:categories)
+                       .where("categories.title = '#{reward_params[:category]}'")
+                       .references(:categories)
+               else
+                 Reward.all.includes(:categories)
+               end
+    @rewards = @rewards.page(params[:page])
+    @categories = Category.pluck(:title)
```

Comment:
Just a detail, but `@categories` name can be a little misleading  - one can think that it stores array of Category objects. Maybe `@category_titles` would be more relevant?

Reply:
Thanks for comments.

You are totally right.